### DISCUSSION TITLE
API to write monochrome BMP image

### DISCUSSION THREAD

**Discussion by rmshub** [ImageMagick/ImageMagick#6410] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

Linux

### Description

Do we have any MagickWand API to set bit depth of BMP image while writing ? The bit depth of image is changed to 24 bit after MagickResizeImage

**Comment by snibgo** [comment]
A quick search of magick-image.c finds `MagickSetImageDepth()`.

**Comment by fmw42** [comment]
Post your input image.  If it was 8-bit palette colored and you resize, you could create more colors than permitted by 8-bits. So the result will be converted to 24-bit true color.

**Comment by rmshub** [comment]
@snibgo `MagickSetImageDepth(wand, 1)` is not working. Making image full white. Test image #
[BMPimage.zip](https://github.com/ImageMagick/ImageMagick/files/11765583/BMPimage.zip)

**Comment by snibgo** [comment]
What else does your program do? I suspect it doesn't simply read a BMP and write a BMP.

I suggest you show the complete program.

**Comment by rmshub** [comment]
Code to test the BMP image resize
```
MagickWandGenesis();
MagickWand *rose = NewMagickWand();
MagickBooleanType status = MagickReadImage(rose, "/sdcard/24b.bmp");
if (status == MagickFalse)
{
    printf("MagickWand: 24b image read failed");
}
else
{ 
    MagickResizeImage(rose, 96, 96, BoxFilter);
    MagickWriteImage(rose, "/sdcard/24b_96_96.bmp");
    printf("24b.bmp (%lu x %lu)\n", MagickGetImageWidth(rose), MagickGetImageHeight(rose));
}

rose = DestroyMagickWand(rose);
MagickWandTerminus();
```

**Comment by snibgo** [comment]
Ah, so you are resizing the image, with a box filter. The input image 24b.bmp is white, with a single diagonal line of black pixels. After the resize, it is white with a single diagonal line of pixels that are about 50.1961% gray.

When you simplify the image to 1-bit, black or white, the 50.1961% gray pixels become white. Hence, the output is entirely white.

What output do you want?

**Comment by rmshub** [comment]
I want single diagonal line of black pixels after resize with 1-bit depth

**Comment by snibgo** [comment]
I suggest the equivalent of:
```
magick 24b.bmp -filter box -resize 96x96 -threshold 75% -colorspace Gray -type bilevel out.bmp
```
